src/pages/ManageStaff/ManageStaff.jsx

```diff
@@ -26,42 +30,77 @@ import UserTable from '../../components/UserTable/UserTable';
 const { SUPERADMIN_ROLE, DRIVER_ROLE, ADMIN_ROLE } = AUTH_ROLES.AUTH_ROLES;
 
 const ManageStaff = ({ cookies }) => {
-  const [users, setUsers] = useState([]);
-  const [usersCopy, setUsersCopy] = useState([]);
+  const [displayedUsers, setDisplayedUsers] = useState([]);
+  const [allUsers, setAllUsers] = useState([]);
+  const [adminUsers, setAdminUsers] = useState([]);
+  const [driverUsers, setDriverUsers] = useState([]);
   const [isSuperAdmin, setIsSuperAdmin] = useState(false);
+  const [currFilter, setCurrFilter] = useState('all');
+
+  const fuse = new Fuse(allUsers, {
+    keys: ['firstName', 'lastName', 'email'],
+  });
+
+  const updateDisplay = () => {
+    if (currFilter === 'all') {
+      setDisplayedUsers(allUsers);
+      fuse.setCollection(allUsers);
+    } else if (currFilter === 'admin') {
+      setDisplayedUsers(adminUsers);
+      fuse.setCollection(adminUsers);
+    } else {
+      setDisplayedUsers(driverUsers);
+      fuse.setCollection(driverUsers);
+    }
+  };
 
   const refreshData = async () => {
-    const { uid } = await getCurrentUser(auth);
-    const currentUser = await getUserFromDB(uid);
-    const userId = currentUser.id;
+    const userId = getCurrentUserId();
```

Comment:
🚫 **[eslint]** <[no-undef](https://eslint.org/docs/rules/no-undef)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'getCurrentUserId' is not defined.